### DISCUSSION TITLE
DynamoDBによるボトルネック解消事例

### DISCUSSION THREAD

**Discussion by Kiyo510** [Progaku-copy/isucon-knowledge#3] [ISUCON本第１章 輪読会]
### 事例
ある通知テーブルに数十億レコードあるテーブルがあり、コストとクエリにとても時間がかかっていた。
RDB→DynamoDBへ起き換えてコストもレイテンシも抑えらえた

## DynamoDBについて

https://www.sbbit.jp/article/cont1/95515

### 注意
- DynamoDBは後からパーティションキーを変更できないのでちゃんと設計しないと破綻するので難しい
  - DynamoDBをはじめとするNoSQLではRDSのようにすべてのカラムに対してWhere句で検索するといったことができない(条件指定は基本的に、プライマリキー以外は使用できない)ので、ちゃんとユースケースを想定してテーブル設計しないと破綻する。

**Comment by kakeru-one** [comment]
ログとかを持たせるならこいつも良いかも
https://dev.classmethod.jp/articles/comparison-dynamodb-and-timestream/

**Comment by Kiyo510** [comment]
- チューニングがんばらなくても札束でも殴れる
  - https://aws.amazon.com/jp/blogs/news/join-the-preview-amazon-aurora-limitless-database/
